File: income-outcome-tracker/models/transaction.go
package models

type Transaction struct {
	TransactionID string  `json:"transaction_id"`
	UserID        string  `json:"user_id"`
	Amount        float64 `json:"amount"`
	Type          string  `json:"type"`
	Date          string  `json:"date"`
	Description   string  `json:"description"`
}

File: express-mongodb/_src/models/transaction.js
const mongoose = require('mongoose');

const TransactionSchema = new mongoose.Schema({
    user_id: {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'User',
        required: true
    },
    amount: {
        type: Number,
        required: true
    },
    type: {
        type: String,
        enum: ['income', 'expense'],
        required: true
    },
    date: {
        type: Date,
        required: true
    },
    description: {
        type: String
    }
});

module.exports = mongoose.model('Transaction', TransactionSchema);